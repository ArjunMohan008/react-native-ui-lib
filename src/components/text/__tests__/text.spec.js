import React from 'react';
import renderer from 'react-test-renderer';
import Text from '../index';

describe('Text', () => {
  it('should render clean text', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('style modifiers', () => {
    it('should render text with red50 as a color', () => {
      const tree = renderer.create(<Text red50 />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render text with white color modifier', () => {
      const tree = renderer.create(<Text bg-white />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render text with text30 typography modifier', () => {
      const tree = renderer.create(<Text text30 />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('spacings modifiers', () => {
    // Margins
    it('should render text with right and left margin', () => {
      const tree = renderer.create(<Text marginR-20 marginL-10 />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render text with top and bottom margin', () => {
      const tree = renderer.create(<Text marginR-20 marginL-10 />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render text with horizontal margin', () => {
      const tree = renderer.create(<Text marginH-20 />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render text with vertical margin', () => {
      const tree = renderer.create(<Text marginV-20 />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
