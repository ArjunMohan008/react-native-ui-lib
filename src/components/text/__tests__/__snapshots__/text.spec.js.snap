// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render clean text 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
/>
`;

exports[`Text spacings modifiers should render text with horizontal margin 1`] = `
<Text
  marginH-20={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "marginHorizontal": 20,
      },
    ]
  }
/>
`;

exports[`Text spacings modifiers should render text with right and left margin 1`] = `
<Text
  marginL-10={true}
  marginR-20={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "marginLeft": 10,
        "marginRight": 20,
      },
    ]
  }
/>
`;

exports[`Text spacings modifiers should render text with top and bottom margin 1`] = `
<Text
  marginL-10={true}
  marginR-20={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "marginLeft": 10,
        "marginRight": 20,
      },
    ]
  }
/>
`;

exports[`Text spacings modifiers should render text with vertical margin 1`] = `
<Text
  marginV-20={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "marginVertical": 20,
      },
    ]
  }
/>
`;

exports[`Text style modifiers should render text with red50 as a color 1`] = `
<Text
  red50={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "color": "#F79A94",
      },
    ]
  }
/>
`;

exports[`Text style modifiers should render text with text30 typography modifier 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "fontFamily": undefined,
        "fontSize": 36,
        "fontWeight": "200",
        "lineHeight": 46,
      },
    ]
  }
  text30={true}
/>
`;

exports[`Text style modifiers should render text with white color modifier 1`] = `
<Text
  bg-white={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
/>
`;
